src/internal/fiberRuntime.ts

```diff
@@ -2225,7 +2225,7 @@ export const mergeAll = dual<
     readonly batching?: boolean | "inherit" | undefined
   }) => Effect.Effect<R, E, Z>
 >(
-  (args) => Predicate.isIterable(args[0]),
+  (args) => Predicate.isFunction(args[2]),
```

Comment:
This will fail if Z is a function :) (even though that is a rare case)

Predicate.isIterable(args[0]) && Predicate.isFunction(args[2]) should cover all the cases.

Reply:
I think that `args[2]` can only be:

- a function `f: (z: Z, a: A, i: number) => Z` (data-first) 
- an object or `undefined` (`options?: { ... }`) (data-last)

Reply:
Ah yes, nice 👍🏼